pkg/api/v1/router.go

```diff
@@ -10,49 +10,99 @@ import (
-
-	rg.GET("/servers/:uuid/attributes", amw.AuthRequired(), amw.RequiredScopes(readScopes("server", "server:attributes")), r.serverAttributesList)
-	rg.POST("/servers/:uuid/attributes", amw.AuthRequired(), amw.RequiredScopes(createScopes("server", "server:attributes")), r.serverAttributesCreate)
-	rg.GET("/servers/:uuid/attributes/:namespace", amw.AuthRequired(), amw.RequiredScopes(readScopes("server", "server:attributes")), r.serverAttributesGet)
-	rg.PUT("/servers/:uuid/attributes/:namespace", amw.AuthRequired(), amw.RequiredScopes(updateScopes("server", "server:attributes")), r.serverAttributesUpdate)
-	rg.DELETE("/servers/:uuid/attributes/:namespace", amw.AuthRequired(), amw.RequiredScopes(deleteScopes("server", "server:attributes")), r.serverAttributesDelete)
-
-	rg.GET("/servers/components", amw.AuthRequired(), amw.RequiredScopes(readScopes("server:component")), r.serverComponentList)
-	rg.POST("/servers/:uuid/components", amw.AuthRequired(), amw.RequiredScopes(createScopes("server", "server:component")), r.serverComponentsCreate)
-	rg.GET("/servers/:uuid/components", amw.AuthRequired(), amw.RequiredScopes(readScopes("server", "server:component")), r.serverComponentGet)
-	rg.PUT("/servers/:uuid/components", amw.AuthRequired(), amw.RequiredScopes(updateScopes("server", "server:component")), r.serverComponentUpdate)
-
-	rg.GET("/servers/:uuid/versioned-attributes", amw.AuthRequired(), amw.RequiredScopes(readScopes("server", "server:versioned-attributes")), r.serverVersionedAttributesList)
-	rg.POST("/servers/:uuid/versioned-attributes", amw.AuthRequired(), amw.RequiredScopes(createScopes("server", "server:versioned-attributes")), r.serverVersionedAttributesCreate)
-	rg.GET("/servers/:uuid/versioned-attributes/:namespace", amw.AuthRequired(), amw.RequiredScopes(readScopes("server", "server:versioned-attributes")), r.serverVersionedAttributesGet)
-
-	rg.GET("/server-component-types", amw.AuthRequired(), amw.RequiredScopes(readScopes("server-component-types")), r.serverComponentTypeList)
-	rg.POST("/server-component-types", amw.AuthRequired(), amw.RequiredScopes(updateScopes("server-component-types")), r.serverComponentTypeCreate)
-
-	rg.GET("/server-component-firmwares", amw.AuthRequired(), amw.RequiredScopes(readScopes("server")), r.serverComponentFirmwareList)
-	rg.POST("/server-component-firmwares", amw.AuthRequired(), amw.RequiredScopes(createScopes("server")), r.serverComponentFirmwareCreate)
-	rg.GET("/server-component-firmwares/:uuid", amw.AuthRequired(), amw.RequiredScopes(readScopes("server")), r.serverComponentFirmwareGet)
-	rg.PUT("/server-component-firmwares/:uuid", amw.AuthRequired(), amw.RequiredScopes(updateScopes("server")), r.serverComponentFirmwareUpdate)
-	rg.DELETE("/server-component-firmwares/:uuid", amw.AuthRequired(), amw.RequiredScopes(deleteScopes("server")), r.serverComponentFirmwareDelete)
+	// require all calls to have auth
+	rg.Use(amw.AuthRequired())
+
+	// /servers
```

Comment:
I don't quite see that much value in the separation as done below. In the format that was before it was easier to figure out what the full path of the object was. Now we have to scroll up to see that.